### PR TITLE
Username overlaps Done button

### LINKED ISSUE #4393: Username overlaps Done button
### Description

*Type*: Bug

*Summary*: Long username overlaps done button. Instead, name should have dynamic font that gets smaller if there is no space.

#### Expected behavior
Username does not overlap other texts.

### Reproduction

- Open Status
- Set long username
- Open contact code sharing screen and check the titlebar

### Additional Information
* Status version: nightly, May 23
* Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by tbenr**
Fixes #4393

### CODE REVIEW

**Comment by tbenr** [comment]
The proposed solution is to change overall toolbar behaviour to have a oneline only title and:

- Android: `ellipsizeMode "tail"`
- iOS: `adjustsFontSizeToFit true`

I propose a different behaviour for the platform because font autoadjustment is supported only for iOS: https://facebook.github.io/react-native/docs/0.54/text#adjustsfontsizetofit

moreover I added a fifth optional argument to `content-title` function so it is possible to add\override text properties.

<blockquote><div><strong><a href="https://facebook.github.io/react-native/index.html">Text · React Native</a></strong></div><div>A React component for displaying text.</div></blockquote>

File: src/status_im/ui/components/toolbar/view.cljs [hunk 1]
```diff
@@ -79,12 +79,16 @@
...
+  ([title-style title subtitle-style subtitle additional-text-props]
    [react/view {:style styles/toolbar-title-container}
-    [react/text {:style (merge styles/toolbar-title-text title-style)
-                 :font  :toolbar-title}
-     title]
+    (if platform/ios?
+      [react/text (merge {:style (merge styles/toolbar-title-text title-style)
+                          :font  :toolbar-title :adjustsFontSizeToFit true :numberOfLines 1}
+                         additional-text-props) title]
+      [react/text (merge {:style (merge styles/toolbar-title-text title-style)
```
Comment (flexsurfer) [inline src/status_im/ui/components/toolbar/view.cljs]:
```
[react/text (merge {:style (merge styles/toolbar-title-text title-style)
                                :font  :toolbar-title :numberOfLines 1}
                                (if platform/ios? {:adjustsFontSizeToFit true} {:ellipsizeMode :tail})
                                 additional-text-props) title]
```

**Review by flexsurfer — CHANGES_REQUESTED** [review]
thanks @tbenr for your contribution i left small comment

File: src/status_im/ui/components/toolbar/view.cljs (hunk 1 above)
Comment (tbenr) [reply src/status_im/ui/components/toolbar/view.cljs]: yes, more compact version. I'll update it shortly.

**Comment by tbenr** [comment]
@flexsurfer I updated the commit, but now jenkins' build is failing. Is there something that I can do to fix or is a jenkins' issue (desktop\develop branch changes)?

File: src/status_im/ui/components/toolbar/view.cljs [hunk 2]
```diff
@@ -79,12 +79,13 @@
 (defn content-title
   ([title] (content-title nil title))
   ([title-style title]
-   (content-title title-style title nil nil))
-  ([title-style title subtitle-style subtitle]
+   (content-title title-style title nil nil nil))
+  ([title-style title subtitle-style subtitle additional-text-props]
```
Comment (flexsurfer) [inline src/status_im/ui/components/toolbar/view.cljs]: what if i call `(content-title title-style title subtitle-style subtitle)` ?

**Comment by flexsurfer** [comment]
hey @tbenr thanks, try to rebase onto develop

File: src/status_im/ui/components/toolbar/view.cljs (hunk 2 above)
Comment (tbenr) [reply src/status_im/ui/components/toolbar/view.cljs]: it should then fallback to `(content-title title-style title subtitle-style subtitle nil)` which then execute the function without altering the prop hashmap, right?

Comment (flexsurfer) [reply src/status_im/ui/components/toolbar/view.cljs]: yes we need to add `[title-style title subtitle-style subtitle]` here

Comment (tbenr) [reply src/status_im/ui/components/toolbar/view.cljs]: ahh yes, I forgot to implement the 4 param version to be retrocompatible... i'll updating asap...

File: src/status_im/ui/components/toolbar/view.cljs [hunk 3]
```diff
@@ -79,12 +79,15 @@
 (defn content-title
   ([title] (content-title nil title))
   ([title-style title]
-   (content-title title-style title nil nil))
-  ([title-style title subtitle-style subtitle]
+   (content-title title-style title nil nil nil))
+  ([title-style title subtitle-style]
```
Comment (flexsurfer) [inline src/status_im/ui/components/toolbar/view.cljs]: `subtitle` missed here ,`[title-style title subtitle-style subtitle]`

Comment (tbenr) [reply src/status_im/ui/components/toolbar/view.cljs]: inadvertence :-1:

Comment (flexsurfer) [reply src/status_im/ui/components/toolbar/view.cljs]: @tbenr any updates on this?

Comment (tbenr) [reply src/status_im/ui/components/toolbar/view.cljs]: @flexsurfer should be ok now.

**Review by flexsurfer — APPROVED** [review]

**Review by dmitryn — APPROVED** [review]

**Comment by lukaszfryc** [comment]
@tbenr it's collapsed on Android but it still overlaps with "Done" button on iOS. I'm testing on iPhone 8, 11.4. Can you recheck on your side?

**Comment by tbenr** [comment]
@lukaszfryc thanks for the feedback. I need time to check, I'm on vacation. In september I should  also get a mac so I'll be finally able to test iOS builds.

**Comment by lukaszfryc** [comment]
@tbenr the issue is still present on iOS:



Could you post a comment with screenshots from at least 1 real Android and iOS device when you apply the final fix?

**Comment by tbenr** [comment]
Hi @lukaszfryc - I'm still not able to test on a iOS (in September I'll get a mac). Looking your screenshot it seems that adding ellipses is not enough. Please put it on hold since I'll be able to test iOS builds.

**Comment by tbenr** [comment]
@lukaszfryc I'm trying to test without a Mac, on a physical iOS (ipad). But it seems you need to add my device UUID to the ios Build to let me install CI builds (correct?). Is it possible? Who I need to contact?

**Comment by lukaszfryc** [comment]
@rasom could you add @tbenr's UUID?

**Comment by tbenr** [comment]
UDID: dedbc41cd5322dc2a84f7ff7a37e37b903dc0010

**Comment by rasom** [comment]
udid is added, next build should be installable

**Comment by rasom** [comment]
started build

**Comment by tbenr** [comment]
@rasom don't seems that **dedbc41cd5322dc2a84f7ff7a37e37b903dc0010** is in the provisioned list of this last build. Can you check?

**Comment by vitvly** [comment]
@tbenr @rasom was `dedbc41cd5322dc2a84f7ff7a37e37b903dc0010` properly added to the build?

**Comment by tbenr** [comment]
@siphiuel  as far as i saw, latest build here was not including the udid. Even if I should get a mac this week, would be useful to have udid added enable me to test on a real device.

**Comment by tbenr** [comment]
@lukaszfryc it should be fixed now. I tested on iOS simulator. Btw even the last build does not contain my UDID, so i cant test on my real iPad yet.

**Comment by lukaszfryc** [comment]
@tbenr looks good now!

**Pull request merged**